src/pages/NotFound.tsx

```diff
@@ -1,15 +1,48 @@
 import React from "react";
 import { useNavigate } from "react-router-dom";
+import Button from "../components/ui/Button";
+import { ButtonSize } from "../utils/types";
+import { useGetProfileQuery } from "../features/user/apiSlice";
+import { getJWT } from "../utils/helper";
 
 const NotFound = () => {
   const navigate = useNavigate();
+  const jwt = getJWT();
+  const {data: profile} = useGetProfileQuery(jwt);
+  const role = profile?.role;
+
+  
+  const navigateBasedOnRole = (role: string) => { 
+    switch (role) {
+      case "Admin":
+        navigate("/forms");
+        break;
+      case "Coach":
+        navigate("/overview");
+        break;
+      case "Applicant" || "Prospect":
```

Comment:
This case will always be "Applicant" since any non-empty string is truthy. So if the role is "Prospect" the default case will be executed instead of this case.

Suggestion: since here you're not checking for validation, you can redirect to `/home` in the default case and get rid of the below part (redirecting to `/login` will be implemented in the part of the application where authentication is being handled)